model/task_queue_service_dependency.go

```diff
@@ -175,17 +176,17 @@ func (t *basicCachedDAGDispatcherImpl) rebuild(items []TaskQueueItem) error {
 		sort.SliceStable(su.tasks, func(i, j int) bool { return su.tasks[i].GroupIndex < su.tasks[j].GroupIndex })
 	}
 
-	// Add edges for task dependencies
 	for _, item := range items {
 		for _, dependency := range item.Dependencies {
-			if err := t.addEdge(item.Id, dependency); err != nil {
+			// addEdge(A, B) means that B depends on A.
+			if err := t.addEdge(dependency, item.Id); err != nil {
 				return errors.Wrapf(err, "failed to create in-memory task queue of TaskQueueItems for distro '%s'; error defining a DirectedGraph incorporating task dependencies", t.distroID)
 			}
 		}
 	}
 
 	// Sort the graph. Use a lexical sort to resolve ambiguities, because node order is the order that we received these in.
-	sorted, err := topo.SortStabilized(t.graph, lexicalReversed)
+	sorted, err := topo.SortStabilized(t.graph, nil)
```

Comment:
the comment is now out of date, and I'm not really sure I understand this change

Reply:
Removed.